crates/server/src/datastore/postgres/mod.rs

```diff
@@ -695,10 +696,12 @@ impl DataStore for PostgresDataStore {
             .first::<Checkpoint>(&mut conn)
             .await?;
 
+        let log_length = checkpoint.log_length.try_into().unwrap();
```

Comment:
Hopefully this code will be long gone by the time anyone overflows a u32-worth of records but I'd still rather blow up if it happens.